src/main/java/bi/deep/HashcodeRegistry.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *    Copyright 2023 Deep BI, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package bi.deep;
+
+import org.apache.druid.segment.DimensionSelector;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class HashcodeRegistry {
+    private static final String NULL = "NULL";
+    private final Set<Integer> hashcodeSet;
+
+    public HashcodeRegistry(Set<Integer> hashcodeSet) {
+        this.hashcodeSet = hashcodeSet;
+    }
+
+    public void add(List<DimensionSelector> selectors) {
+        hashcodeSet.add(getHashcode(selectors));
+    }
+
+    public int size() {
+        return hashcodeSet.size();
+    }
+
+    public boolean contains(List<DimensionSelector> selectors) {
+        return hashcodeSet.contains(getHashcode(selectors));
+    }
+
+    private int getHashcode(List<DimensionSelector> selectors) {
```

Comment:
@BartMiki 
https://preshing.com/20110504/hash-collision-probabilities/#:~:text=a%2050%25%20chance%20of%20collision%20occurs%20when%20the%20number%20of%20hashes%20is%2077163

We could store the objects instead of hashcodes i'm not sure that it will be a good way to solve it, or maybe  we can count our own Long hashCode to minimize collision chanses?

Reply:
Chances are highly unlikely given inbuilt hashcode method use good distribution mechanism,  but still there are chances. 
As @nozjkoitop mentioned we could work on some approach to work to minimize hashcode collision. We can also store value if we are ok with memory usage.

Reply:
If we store value then it would be exact, I will ask Michał to confirm, but this is always a tradeoff